### PR TITLE
Feature/add details view

### PR DESCRIPTION
**Pull request by marcinreszko**

### CODE REVIEW

File: src/views/main/MainView.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { withLazy } from 'shared/utils';
 import { Navbar, Sidebar } from '.';
 
 import csx from './MainView.scss';
+import TemplateDetailsView from 'viewstemplates/template-details';
```
Comment (polubis) [inline src/views/main/MainView.tsx:9]: Use withLazy HOC for dynamically loading this code. Also change path to views/template-details.

File: src/views/templates/index.ts [hunk 2]
```diff
@@ -3,4 +3,5 @@ import TemplatesView from './TemplatesView';
 export * from './search-categories';
 export * from './template-search';
 export * from './template-tiles';
+export * from './template-details';
```
Comment (polubis) [inline src/views/templates/index.ts:6]: Template-details is separate view so should be located directly after views/template-details and should have structure like templates-view. 
So structure should look like:
views
   templates
   login
   template-details - here put your code

File: src/views/templates/template-details/TemplateDetailsView.scss [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  @include centered;
+  height: 100%;
+  .container {
+    @include colCenter;
+    padding: $padding;
+    width: 50%;
+  }
+}
+
+section {
```
Comment (polubis) [inline src/views/templates/template-details/TemplateDetailsView.scss:18]: This is ultra risky. You making all section markups in app styled like that. Put all code under .container. CSS modules don't use shadow DOM like in Angular. They creating unique class names. Use class instead if you want style it like that

File: src/views/templates/template-details/TemplateDetailsView.scss [hunk 4]
```diff
@@ -0,0 +1,123 @@
+@import 'styles/mixins';
+@import 'styles/palette';
+
+$padding: 20px 0;
+
+
+/* layout */
+.root {
```
Comment (polubis) [inline src/views/templates/template-details/TemplateDetailsView.scss:8]: Rename this to templateDetailsView - sometime ago i talked about root classes which is convention from Material UI. After testing this - i think we should use normal class names which tells more. templateDetailsView will be good

File: src/views/templates/template-details/TemplateDetailsView.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
+import React, { useEffect } from 'react';
+import csx from './TemplateDetailsView.scss';
+import { Button } from 'shared/ui';
+
+import EditIcon from '@material-ui/icons/Edit';
+import StarBorderIcon from '@material-ui/icons/StarBorder';
```
Comment (polubis) [inline src/views/templates/template-details/TemplateDetailsView.tsx:6]: Create segments from imports. At top add all core imports like react-router, react. Add enter and then material ui imports, add enter and shared imports, add enter and move csx imports from top to bottom. You can check already created files.

File: src/views/templates/template-details/TemplateDetailsView.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import VisibilityIcon from '@material-ui/icons/Visibility';
+import MenuBookIcon from '@material-ui/icons/MenuBook';
+import ShareIcon from '@material-ui/icons/Share';
+
+import { RouteChildrenProps } from 'react-router';
+
+interface TemplateDetailsViewProps extends RouteChildrenProps<{ id: string }> {}
+
+export const TemplateDetailsView = ({ match }: TemplateDetailsViewProps) => {
```
Comment (polubis) [inline src/views/templates/template-details/TemplateDetailsView.tsx:15]: This should be created as default export. All components with View postfix should be exported default. Check TemplatesView for reference

File: src/views/templates/template-details/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { TemplateDetailsView } from './TemplateDetailsView';
```
Comment (polubis) [inline src/views/templates/template-details/index.ts:1]: Should be import TemplateView without brackets.

File: src/views/templates/index.ts (hunk 2 above)
Comment (marcinreszko) [reply src/views/templates/index.ts:6]: Done

**Pull request merged**